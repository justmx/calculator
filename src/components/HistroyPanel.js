import React from 'react';

const HistroyPanel = ({records = [], clearAll}) => {
	return (
		<div className='histrory-container'>
  			<div className='histrory-container-header'>
  				<h3>History</h3>
  				<button className='clear-button' onClick={() => clearAll()}> Clear All History </button>
  			</div>
  			<div className='histrory-container-inner'>
  				{
  					records.map((list, i) => {
  						return (<p className='record-item' key={i}>{list.replace('*','x')}</p>)
  					})
  				}
  			</div>  		
  		</div>
	)
}

export default HistroyPanel;
