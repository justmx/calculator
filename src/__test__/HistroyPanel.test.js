import React from 'react';
import { shallow } from 'enzyme';
import HistroyPanel from '../components/HistroyPanel';

const records = [
  '1 + 1 = 2', '1 + 2 = 3', '1 x 1 = 1'
]

describe('HistroyPanel', () => {
  it('should render HistroyPanel', () => {
    const histroyPanel = shallow(<HistroyPanel records={records} />);
    expect(histroyPanel).toMatchSnapshot();
  });

  it('renders record items', () => {
    const histroyPanel = shallow(<HistroyPanel records={records} />);
    expect(histroyPanel.find('.record-item')).toBeDefined();
    expect(histroyPanel.find('.record-item')).toHaveLength(records.length);
  });
});