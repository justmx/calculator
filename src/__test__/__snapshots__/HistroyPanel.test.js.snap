// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistroyPanel should render HistroyPanel 1`] = `
<div
  className="histrory-container"
>
  <div
    className="histrory-container-header"
  >
    <h3>
      History
    </h3>
    <button
      className="clear-button"
      onClick={[Function]}
    >
       Clear All History 
    </button>
  </div>
  <div
    className="histrory-container-inner"
  >
    <p
      className="record-item"
      key="0"
    >
      1 + 1 = 2
    </p>
    <p
      className="record-item"
      key="1"
    >
      1 + 2 = 3
    </p>
    <p
      className="record-item"
      key="2"
    >
      1 x 1 = 1
    </p>
  </div>
</div>
`;
